odoo/addons/base/models/ir_mail_server.py

```diff
@@ -275,17 +303,28 @@ def test_smtp_connection(self):
                     # ignored, just a consequence of the previous exception
                     pass
 
-        message = _("Connection Test Successful!")
+        messages = [_('Connection Test Successful!')]
+        if autodetect_max_email_size:
+            if self.max_email_size:
+                messages.append(_('Email maximum size updated to %(size)s MB',
+                                  size=', '.join(f"{server.max_email_size:.2f}" for server in self)))
+            else:
+                raise UserError(_("The server doesn't return the maximum email size."))
         return {
             'type': 'ir.actions.client',
             'tag': 'display_notification',
             'params': {
-                'message': message,
+                'message': ' '.join(messages),
                 'type': 'success',
                 'sticky': False,
-            }
+            },
         }
 
+    def action_retrieve_max_email_size(self):
+        self.ensure_one()
+        # Don't return the client notification because otherwise the data are not updated on the client side
```

Comment:
I think this is possible to do both, I've done it recently in: https://github.com/odoo/odoo/blob/master/addons/event_crm/models/event_event.py#L75

(If I understand this comment properly).